src/Sharpliner/AzureDevOps/TemplateDefinition.cs

```diff
@@ -257,3 +260,79 @@ internal TemplateDefinition()
     {
     }
 }
+
+public abstract class TemplateDefinition<T, TParameters> : TemplateDefinition<T> where TParameters : class, new()
+{
+    protected TParameters TypedParameters { get; private set; }
+
+    protected TemplateDefinition()
+    {
+        TypedParameters = new();
+    }
+
+    protected TemplateDefinition(TParameters typedParameters)
+    {
+        TypedParameters = typedParameters;
+    }
+
+    public sealed override List<Parameter> Parameters => ToParameters(TypedParameters);
+
+    public TemplateParameters TemplateParameters => ToTemplateParameters(TypedParameters);
```

Comment:
Do we need this?

Reply:
we can probably inline this call in `public static implicit operator Template<T>(TemplateDefinition<T, TParameters> definition)`

Reply:
cleaned up